packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/bedrock/create-model-customization-job.ts

```diff
@@ -0,0 +1,515 @@
+   * @see https://docs.aws.amazon.com/AWSEC2/latest/APIReference/Run_Instance_Idempotency.html
+   *
+   * @default - no client request token
+   */
+  readonly clientRequestToken?: string;
+
+  /**
+   * The customization type.
+   *
+   * @default FINE_TUNING
+   */
+  readonly customizationType?: CustomizationType;
+
+  /**
+   * The custom model is encrypted at rest using this key.
+   *
+   * @default - no encryption
+   */
+  readonly customModelKmsKey?: kms.IKey;
+
+  /**
+   * A name for the resulting custom model.
+   *
+   * The maximum length is 63 characters and it needs to satisfy the regular expression ^([0-9a-zA-Z][_-]?)+$.
+   */
+  readonly customModelName: string;
+
+  /**
+   * Tags to attach to the resulting custom model.
+   *
+   * The maximum number of tags is 200.
+   *
+   * @default - no tags
+   */
+  readonly customModelTags?: ITag[];
+
+  /**
+   * Parameters related to tuning the model.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/userguide/custom-models-hp.html
+   *
+   * @default - use default hyperparameters
+   */
+  readonly hyperParameters?: { [key: string]: string };
+
+  /**
+   * A name for the fine-tuning job.
+   *
+   * The maximum length is 63 characters and it needs to satisfy the regular expression ^[a-zA-Z0-9](-*[a-zA-Z0-9\+\-\.])*$.
+   */
+  readonly jobName: string;
+
+  /**
+   * Tags to attach to the job.
+   * The maximum number of tags is 200.
+   *
+   * @default - no tags
+   */
+  readonly jobTags?: ITag[];
+
+  /**
+   * The S3 bucket configuration where the output data is stored.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_OutputDataConfig.html
+   */
+  readonly outputData: BucketConfiguration;
+
+  /**
+   * The IAM role that Amazon Bedrock can assume to perform tasks on your behalf.
+   *
+   * For example, during model training, Amazon Bedrock needs your permission to read input data from an S3 bucket,
+   * write model artifacts to an S3 bucket.
+   * To pass this role to Amazon Bedrock, the caller of this API must have the iam:PassRole permission.
+   *
+   * @default - use auto generated role
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * The S3 bucket configuration where the training data is stored.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_TrainingDataConfig.html
+   */
+  readonly trainingData: BucketConfiguration;
+
+  /**
+   * The S3 bucket configuration where the validation data is stored.
+   *
+   * The maximum number is 10.
+   *
+   * @see https://docs.aws.amazon.com/bedrock/latest/APIReference/API_Validator.html
+   */
+  readonly validationData: BucketConfiguration[];
+
+  /**
+   * Configuration parameters for the private Virtual Private Cloud (VPC) that contains the resources you are using for this job.
+   *
+   * @default - no VPC configuration
+   */
+  readonly vpcConfig?: IBedrockCreateModelCustomizationJobVpcConfig;
```

Comment:
```suggestion
   * The minimum number of security groups for the VPC is 1.
   * The maximum number of security groups for the VPC is 5.
   *
   * The maximum number of subnets for the VPC is 1.
   * The maximum number of subnets for the VPC is 16.
   *
   * @default - no VPC configuration
   */
  readonly vpc?: ec2.IVpc;
```
This might simplify things in case we want to add the optional VPC endpoint policy and bucket policy (requires the VPC ID) in the future. Also, it provides a cleaner interface.
What do you think?

Reply:
Thank you for your suggestion! I'm agree with your suggestion.
While this approach eliminates the ability to specify specific subnets and security groups explicitly, I believe it would be appropriate to address any requests as they arise over time.

My only concern is regarding the settings for the security group. If only the `vpc` is specified as an argument, there will be a need to automatically generate a security group for the ENI used by bedrock. For the outbound rules of this security group, is it acceptable to allow access to S3 via HTTPS to 0.0.0.0/0?

If the strictest outbound rules are required, the settings would be as follows:
- When using an interface endpoint: Allow HTTPS to the security group attached to the S3 interface endpoint.
- When using a gateway endpoint, NAT gateway or assigning public IP to the ENI: Allow HTTPS to the S3 prefix list.

(But I don't want to implement it...)

Reply:
I think Bedrock is executed outside of a VPC. Therefore, Bedrock does not create VPC resources if a VPC configuration is not provided. 
If we want to more strictly manage access to S3 resources, I believe that specifying a VPC configuration will result in the creation of an ENI within the VPC for Bedrock to access. 

Please feel free to correct me if my understanding is incorrect.

Reply:
>Bedrock does not create VPC resources if a VPC configuration is not provided.

So if you provide `vpcConfig` Bedrock create a new VPC?

>Please feel free to correct me if my understanding is incorrect.

I'm trying to understand how it works myself, the documentation doesn't make it clear. Can you test it out and see if providing `vpcConfig` creates a new VPC?

Reply:
@comcalvi Sorry for the unclear sentence. 
I believe Bedrock does not create a new VPC but instead creates a new ENI within the existing VPC. When a vpcConfig is provided, Bedrock searches for an existing VPC and creates an ENI in the specified subnet, then attaches the provided security group.

Reply:
@comcalvi Could you confirm above message?

[Here is a reference document](https://docs.aws.amazon.com/bedrock/latest/userguide/vpc-model-customization.html#vpc-config) for creating an ENI.

> When you specify the VPC subnets and security groups for a job, Amazon Bedrock creates elastic network interfaces (ENIs) that are associated with your security groups in one of the subnets. ENIs allow the Amazon Bedrock job to connect to resources in your VPC.

Reply:
got it, thanks.

Reply:
@badmintoncryer what do you think of implementing the change @lpizzinidev suggested; turning `VpcConfig` into an `IVpc`? Since Bedrock doesn't create one for you, I don't see the harm.

Reply:
@badmintoncryer what do you think of implementing the change @lpizzinidev suggested; turning VpcConfig into an IVpc?

Reply:
@comcalvi 

Changing the argument to `IVpc` is not an issue!

As mentioned above, there is only concern about the security group attached to the ENI.
When using `IVpc` as an argument, a security group needs to be generated automatically. In this case, is it acceptable to allow all outbound rules?